### ISSUE TITLE
MD045 fires on Obsidian embeds (`![[...]]`), which have no alt-text syntax

### ISSUE THREAD

**Issue by rayshan** [rvben/rumdl#782]
Under `flavor = "obsidian"`, MD045 fires on Obsidian embeds (`![[...]]`). Those are not Markdown images, and the syntax has no alt-text slot, so the warning is unactionable — there is no edit that satisfies it.

## Reproduction

`rumdl 0.2.48`, two files in one directory:

`Target.md`

```md
# Target

## A heading

Body text.
```

`Note.md`

```md
# Note

![[Target#A heading]]

![[Target]]

![[diagram.png]]

![](real.png)

![alt](real.png)
```

```console
$ rumdl check . --no-config --config 'global.flavor = "obsidian"'
Note.md:3:1: [MD045] Image missing alt text (add description for accessibility: ![description](url))
Note.md:5:1: [MD045] Image missing alt text (add description for accessibility: ![description](url))
Note.md:7:1: [MD045] Image missing alt text (add description for accessibility: ![description](url))
Note.md:9:1: [MD045] Image missing alt text (add description for accessibility: ![description](url))
```

Lines 3, 5 and 7 are Obsidian embeds. Line 9 is a real Markdown image and a correct hit; line 11 is correctly silent.

One wrinkle that may help locate it: the note embeds are only flagged when the target resolves. With `Target.md` absent, `![[Target#A heading]]` is not flagged, while `![[diagram.png]]` is flagged either way.

## Expected

Lines 3, 5 and 7 produce no MD045. Only line 9 does.

## Why these can't be fixed in the source

**A note embed is a content transclusion, not an image.** [Embed files](https://help.obsidian.md/embeds) documents `![[Internal links]]` and `![[Internal links#^b15695]]` under "Embed a note in another note". The rendered output is the target note's content, so alt text is not a meaningful concept.

**Image embeds have no alt-text syntax.** The pipe in an Obsidian image embed carries *dimensions*: "You can change the image dimensions by adding `|640x480` to the link destination, where 640 is the width and 480 is the height... If you only specify the width, the image scales proportionally. For example, `![[Engelbart.jpg|100]]`." So `![[diagram.png|Some description]]` does not set alt text; it is parsed as a size. The accessibility goal MD045 exists to serve is only reachable in Obsidian by rewriting to a Markdown image, which changes behavior (embeds resolve through the vault's attachment search, relative paths do not).

Currently `![[diagram.png|300]]` silences the rule, which is the wrong outcome in the other direction: a dimension string is being accepted as alt text.

## Suggested fix

Under `flavor = "obsidian"`, exclude `![[...]]` from MD045 entirely, the same way wikilinks were excluded from MD051/MD057 in #235. MD045 exposes no configuration options, so there is no way to opt out per project short of disabling the rule everywhere, which also loses the true positives on real Markdown images.

If distinguishing the two embed kinds is worthwhile, note embeds should be unconditionally skipped, and image embeds skipped as well since no conforming alt text can be written.

**Comment by rvben** [comment]
Thanks @rayshan, reproduced on 0.2.48. Two corrections though, both of which change the fix:

**Not flavor-specific.** All three cases fire under `global.flavor = "standard"` too. `Options::ENABLE_WIKILINKS` is set unconditionally for every flavor (`src/utils/parser_options.rs:26`), so gating the fix on the obsidian flavor would leave the false positive in place for everyone else.

**The "only when the target resolves" wrinkle doesn't reproduce.** With `Target.md` deleted, `![[Target]]` and `![[Target#A heading]]` are still flagged. So it isn't a resolution check.

The cause is `94e9b41f`, which reads the text after the pipe in `![[file.png|...]]` as alt text. Per the Obsidian docs the pipe sets dimensions (`![[Engelbart.jpg|100]]`), and embeds have no alt-text slot at all, so the current behavior is wrong both ways: `![[x.png|300]]` gets silenced by a dimension string, and `![[x.png]]` gets flagged with no edit that can satisfy the rule. Note embeds aren't images in the first place.

Fix is what you propose minus the flavor gate: skip `LinkType::WikiLink` in MD045 for all flavors, and stop treating the pipe as alt text.

Related, found while confirming: MD057 has the same shape, skipping wikilinks in the links loop but not the images loop, so `![[diagram.png]]` reports a missing relative link while `[[diagram.png]]` correctly doesn't. Obsidian resolves attachments vault-wide, so that's a false positive too. I'll fix it alongside.

**Comment by rvben** [comment]
Fixed in v0.2.49.

`![[note]]` and `![[img.png|300]]` are no longer treated as images by MD045 or MD057, and the pipe is no longer read as alt text. Not flavor-gated, so it applies under `standard` too.

Three related bugs surfaced once the constructs were visible, all in the same release: MD044 rewrote the piped dimensions as prose (`![[javascript.png|300]]` to `![[JavaScript.png|300]]`), MD039 turned `[[Target| Display ]]` into `[Display]`, and MD044 rewrote a nested image's URL inside a wikilink's display text (which it had been doing for regular links all along).

**Comment by rayshan** [comment]
Confirmed working. Thanks!

**Comment by rvben** [comment]
Thanks for the confirmation, @rayshan! 💪
